alacritty_terminal/src/renderer/rects.rs

```diff
@@ -110,33 +107,52 @@ impl Rects {
                     && cell.column == line.end.col + 1
                 {
                     // Update the length of the line
-                    line.update_end(cell.into(), size);
-
+                    line.end = cell.into();
                     continue;
                 }
             }
 
             // Start new line if there currently is none
-            let rect = Line::from_cell(cell, *flag, metrics, size);
+            let line = Line { start: cell.into(), end: cell.into(), color: cell.fg };
             match self.inner.get_mut(flag) {
-                Some(lines) => lines.push(rect),
+                Some(lines) => lines.push(line),
                 None => {
-                    self.inner.insert(*flag, vec![rect]);
+                    self.inner.insert(*flag, vec![line]);
                 },
             }
         }
     }
+}
+
+/// Generall purpose rects.
+#[derive(Default)]
+pub struct Rects {
+    inner: Vec<(Rect<f32>, Rgb)>,
+}
+
+impl Rects {
+    pub fn new() -> Self {
```

Comment:
Neither `new` or `default` are used anymore, so I think we can remove the derive and the new impl.

Reply:
Ok, just wanted to save the thing to easy create Rects.

Reply:
Makes sense, but I don't think we need this in our current code and adding it should be simple enough once it actually is required.